indra/util/statement_presentation.py

```diff
@@ -91,100 +517,137 @@ def group_and_sort_statements(stmt_list, ev_totals=None, source_counts=None):
+                agp_m = agent_pair_metrics[ag_key]
+
+                # Check if this statement is a type we ought to skip.
+                is_abbrev_complex = verb == 'Complex' and len(ag_key) <= 2
+                is_abbrev_conv = (verb == 'Conversion'
+                                  and all(isinstance(e, str) for e in ag_key))
+                is_abbrev_stmt = is_abbrev_conv or is_abbrev_complex
+                all_ev_in_rel = rel_m['ev_count'] == agp_m['ev_count']
+                all_have_many_ags = all(_get_ag_name_set_len(s) > 2
+                                        for s in stmts)
+                if is_abbrev_stmt and all_ev_in_rel and all_have_many_ags:
                     continue
-            new_key = (arg_count, inps, sub_count, verb)
-            stmts = sorted(stmts,
-                           key=(lambda s:
-                                _counts(s)[0] + 1 / (1+len(s.agent_list()))),
-                           reverse=True)
-            if source_counts:
-                yield new_key, verb, stmts,\
-                      dict(zip(cols, arg_counts[inps][1:].astype(int))),\
-                      dict(zip(cols, stmt_counts[key][1:].astype(int)))
-            else:
-                yield new_key, verb, stmts
 
-    sorted_groups = sorted(processed_rows(stmt_rows),
-                           key=lambda tpl: tpl[0], reverse=True)
+                sort_key = (_sort_func(agp_m.get_dict()), str(ag_key),
+                            _sort_func(rel_m.get_dict()), str(rel_key))
+
+                stmts = sorted(stmts, key=stmt_sorter, reverse=True)
+
+                yield sort_key, ag_key, rel_key, stmts, agp_m.get_dict(), \
+                    rel_m.get_dict()
+    else:  # If grouped by relation.
+        def processed_rows(stmt_rows):
+            for rel_key, stmts in stmt_rows.items():
+                rel_m = relation_metrics[rel_key]
+                sort_key = (_sort_func(rel_m.get_dict()), str(rel_key))
 
-    return sorted_groups
+                yield sort_key, rel_key, stmts, rel_m.get_dict()
 
+    return sorted(processed_rows(relation_stmts), key=lambda tpl: tpl[0],
+                  reverse=True)
 
-def make_stmt_from_sort_key(key, verb, agents=None):
-    """Make a Statement from the sort key.
+
+def make_stmt_from_relation_key(relation_key, agents=None):
+    """Make a Statement from the relation key.
 
     Specifically, the sort key used by `group_and_sort_statements`.
     """
+
     def make_agent(name):
         if name == 'None' or name is None:
             return None
         return Agent(name)
 
+    verb = relation_key[0]
+    inps = relation_key[1:]
     StmtClass = get_statement_by_name(verb)
-    inps = list(key[1])
     if agents is None:
         agents = []
     if verb == 'Complex':
         agents.extend([make_agent(name) for name in inps])
         stmt = StmtClass(agents[:])
     elif verb == 'Conversion':
+        if isinstance(inps[1], str):
+            pass
```

Comment:
Note to self, figure out what you were thinking here and finish it if necessary.